shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -83,7 +83,18 @@ CGRect ConvertRectToGlobal(SemanticsObject* reference, CGRect local_rect) {
   CGFloat scale = [[[reference bridge]->view() window] screen].scale;
   auto result =
       CGRectMake(rect.x() / scale, rect.y() / scale, rect.width() / scale, rect.height() / scale);
-  return UIAccessibilityConvertFrameToScreenCoordinates(result, [reference bridge]->view());
+  CGRect resultInScreenCoordinates =
+      UIAccessibilityConvertFrameToScreenCoordinates(result, [reference bridge]->view());
+  NSCAssert(isfinite(resultInScreenCoordinates.origin.x) &&
+                !isnan(resultInScreenCoordinates.origin.x) &&
+                isfinite(resultInScreenCoordinates.origin.y) &&
+                !isnan(resultInScreenCoordinates.origin.y) &&
+                isfinite(resultInScreenCoordinates.size.width) &&
+                !isnan(resultInScreenCoordinates.size.width) &&
+                isfinite(resultInScreenCoordinates.size.height) &&
+                !isnan(resultInScreenCoordinates.size.height),
+            @"The global rect must be finite %@", NSStringFromCGRect(resultInScreenCoordinates));
```

Comment:
You should make these separate assert calls.  The way this is written, if one of them fails, you won't know which one failed.

Reply:
Also split out a function `isRealNumber(x) => isfinite(x) && !isnan(x)`